File: 2under_bit.java
import java.util.*;

class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        for(int i=0;i<numbers.length;i++){
            if(numbers[i] == 0){
                answer[i] = 1;
                continue;
            }
            long number = numbers[i];
            List<Long> num = new ArrayList<>();
            
            while(number >= 1){
                num.add(number%2);
                number /= 2;
            }
            num.add(Long.valueOf(0));
            for(int j=0;j<num.size()-1;j++){
                if(j == 0 && num.get(0) == 0){
                    num.set(0,Long.valueOf(1));
                    break;
                }
                if(num.get(j) == 1 && num.get(j+1) == 0){
                    num.set(j,Long.valueOf(0));
                    num.set(j+1,Long.valueOf(1));
                    break;
                }
            }
            long result = 0;
            for(int j=0;j<num.size();j++) result += Math.pow(2,j) * num.get(j);
            answer[i] = result;
        }
        return answer;
    }
}

File: Search/2under_bit.py
def solution(numbers):
	answer = []
	for num in numbers:
		n = []
		if num == 0:
			answer.append(1)
			continue
		while num >= 1:
			n.append(num%2)
			num = num // 2

		n.append(0)
		for i in range(len(n)-1):
			if i == 0 and n[i] == 0:
				n[0] = 1
				break
			elif n[i] == 1 and n[i+1] == 0:
				n[i] = 0
				n[i+1] = 1
				break
		result = 0
		for i in range(len(n)):
			result += (2**(i)) * n[i]
		answer.append(result)


	return answer

if __name__ == "__main__":
	print(solution([1]))